src/tactic/simps.lean

```diff
@@ -39,6 +40,52 @@ setup_tactic_parser
 declare_trace simps.verbose
 declare_trace simps.debug
 
+/-- Projection data for a single projection of a structure, consisting of the following fields:
+  - a custom name
+  - an expression used by simps for the projection. It must be definitionally equal to an original
+    projection (or a composition of multiple projections).
+    These expressions can contain the universe parameters specified in the first argument of
+    `simps_str_attr`.
+  - a list of natural numbers, which is the projection number(s) that have to be applied to the
+    expression. For example the list `[0, 1]` corresponds to applying the first projection of the
+    structure, and then the second projection of the resulting structure (this assumes that the
+    target of the first projection is a structure with at least two projections).
+    The composition of these projections is required to be definitionally equal to the provided
+    expression.
+  - A boolean specifying whether simp-lemmas are generated for this projection by default.
+  - A boolean specifying whether this projection is written as prefix. -/
+@[protect_proj, derive [has_reflect, inhabited]]
+meta structure projection_data :=
+(name : name)
+(expr : expr)
+(proj_nrs : list ℕ)
+(is_default : bool)
+(is_prefix : bool)
+
+/-- Temporary projection data parsed from `initialize_simps_projections` before the expression
+  matching this projection has been found. Only used internally in `simps_get_raw_projections`. -/
+meta structure parsed_projection_data :=
+(orig_name : name) -- original name of the projection
+(new_name : name) -- name used by simps for this projection
```

Comment:
Do I understand the usage of these fields correctly?
```suggestion
(orig_name : name) -- name for this projection used in the structure definition
(new_name : name) -- name for this projection used in the generated `simp` lemma
```

Reply:
Thanks, that is indeed clearer. You got it right.